### PR TITLE
[Audio] Remove players which no longer have a guild.

### PR DESCRIPTION
**Pull request by mikeshardmind**
### Type

- [X] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes

Cleanup players when the bot has one for a guild it leaves.

### CODE REVIEW

**Comment by PredaaA** [comment]
I tested this PR, but I've this error when the bot quit a guild.
```[09/02/2019 14:45] ERROR events on_error 196: Exception in guild_remove
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-packages\discord\client.py", line 230, in _run_event
    await coro(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in on_guild_remove
    for y in {g.voice_channels for g in self.bot.guilds}
  File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in <setcomp>
    for y in {g.voice_channels for g in self.bot.guilds}
TypeError: unhashable type: 'list'
[09/Feb/2019 14:45:29] ERROR - Exception in guild_remove
Traceback (most recent call last):
  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-packages\discord\client.py", line 230, in _run_event
    await coro(*args, **kwargs)
  File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in on_guild_remove
    for y in {g.voice_channels for g in self.bot.guilds}
  File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in <setcomp>
    for y in {g.voice_channels for g in self.bot.guilds}
TypeError: unhashable type: 'list'

**Comment by mikeshardmind** [comment]
> 
> 
> I tested this PR, but I've this error when the bot quit a guild.
> 
> ```
> Traceback (most recent call last):
>   File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-packages\discord\client.py", line 230, in _run_event
>     await coro(*args, **kwargs)
>   File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in on_guild_remove
>     for y in {g.voice_channels for g in self.bot.guilds}
>   File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in <setcomp>
>     for y in {g.voice_channels for g in self.bot.guilds}
> TypeError: unhashable type: 'list'
> [09/Feb/2019 14:45:29] ERROR - Exception in guild_remove
> Traceback (most recent call last):
>   File "C:\Users\Administrator\AppData\Local\Programs\Python\Python37\lib\site-packages\discord\client.py", line 230, in _run_event
>     await coro(*args, **kwargs)
>   File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in on_guild_remove
>     for y in {g.voice_channels for g in self.bot.guilds}
>   File "C:\Users\Administrator\Desktop\cogs_modified\audio\audio.py", line 3195, in <setcomp>
>     for y in {g.voice_channels for g in self.bot.guilds}
> TypeError: unhashable type: 'list'
> ```

:man_facepalming: Fixed.

**Review by PredaaA — COMMENTED** [review]
I have tested it and it works well now.

File: redbot/cogs/audio/audio.py [hunk 1]
```diff
@@ -2667,3 +2667,21 @@ def __unload(self):
...
+            for y in [g.voice_channels for g in self.bot.guilds]
+            for x in y  # y is a list of voice channels
+        }  # Yes, this is ugly. It's also the most performant and commented.
+
+        zombie_players = {p for p in lavalink.player_manager.players if p.channel not in channels}
+        # Do not unroll to combine with next line.
+        # Can result in iterator changing size during context switching.
+        for zombie in zombie_players:
+            await zombie.disconnect()
+            lavalink.player_manager.players.remove(zombie)
```
Comment (Tobotimus) [inline redbot/cogs/audio/audio.py]: You can bump the version of red-lavalink to 0.2.2 in setup.cfg and use `player.destroy()` if you'd like

**Review by Tobotimus — APPROVED** [review]

**Pull request merged**